jrnl/DayOneJournal.py

```diff
@@ -52,7 +52,16 @@ def open(self):
                     except (KeyError, pytz.exceptions.UnknownTimeZoneError):
                         timezone = tzlocal.get_localzone()
                     date = dict_entry["Creation Date"]
-                    date = date + timezone.utcoffset(date, is_dst=False)
+                    try:
```

Comment:
Passing on errors is an anti-pattern I'd like to get us out of whenever possible. 

How about something like this?
```
if dict_entry.get("Time Zone") in pytz.all_timezones:
    timezone = pytz.timezone(dict_entry["Time Zone"])
else:
    timezone = tzlocal.get_localzone()

date = dict_entry["Creation Date"]
if timezone.zone != "UTC":
    date = date + timezone.utcoffset(date, is_dst=False)
```

Reply:
Sorry, apparently didn't hit submit to the above comment yesterday (found it sitting in a tab today).

Reply:
I agree that generally passing on an a raised error is bad from, but what about cases like this where the raised error is actually telling us things are working as expected? This from also supports the philosophy of "it's easier to ask for forgiveness than permission", which is often typically idiomatic Python.

The [article](https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/) I found on this (first link on Google) gives an example and explicitly uses the `except: pass` formation. The [Python docs](https://docs.python.org/3.8/glossary.html#term-eafp) seem to support this form as well.

So I think we should leave it as it is as being more idiomatic Python.

Reply:
@wren : a decision here is what's holding to #786. I like it this way, and think it's more "Pythonic", but of you still want it changed, I'm happy to make the change. Just key me know.

Reply:
Yes, please. Thank you!